In [1]:
import pickle
import json
from sys import getsizeof

## JSON serialize and Binary serialize

In [2]:
my = {'Last Name': 'XI', 'First Name': 'Dongsheng'}

# get size in bytes
print('my type is {}, size is {} bytes'.format(type(my), getsizeof(my)))
my


my type is <class 'dict'>, size is 240 bytes


{'Last Name': 'XI', 'First Name': 'Dongsheng'}

In [3]:
my_serial = pickle.dumps(my)

print('after serialization, my_serial type is {}, size is {} bytes'.format(type(my_serial), getsizeof(my_serial)))
my_serial

after serialization, my_serial type is <class 'bytes'>, size is 99 bytes


b'\x80\x03}q\x00(X\t\x00\x00\x00Last Nameq\x01X\x02\x00\x00\x00XIq\x02X\n\x00\x00\x00First Nameq\x03X\t\x00\x00\x00Dongshengq\x04u.'

In [12]:
my_json = json.dumps(my)

print('after json serialize, my_json type is {}, size is {} bytes'.format(type(my_json), getsizeof(my_json)))
my_json

after json serialize, my_json type is <class 'str'>, size is 95 bytes


'{"Last Name": "XI", "First Name": "Dongsheng"}'

          
|       Type      | Size(Bytes) |         Content                                                                |                |
|:---------------:|-------------|:-------------------------------------------------------------------------------------------------------------------------------------:|----------------|
| <class 'dict'>  | 240         | {'Last Name': 'XI', 'First Name': 'Dongsheng'}                                                                                        |                |
| <class 'bytes'> | 99          | b'\x80\x03}q\x00(X\t\x00\x00\x00Last Nameq\x01X\x02\x00\x00\x00XIq\x02X\n\x00\x00\ x00First Nameq\x03X\t\x00\x00\x00Dongshengq\x04u.' | pickle.dumps() |
| <class 'str'>   | 95          | {"Last Name": "XI", "First Name": "Dongsheng"}                                                                                        | json.dumps()   |

https://code.tutsplus.com/tutorials/serialization-and-deserialization-of-python-objects-part-1--cms-26183

In [11]:
my_serial_deserial = pickle.loads(my_serial)
getsizeof(my_serial_deserial)

240

In [6]:
my_serial_deserial

{'Last Name': 'XI', 'First Name': 'Dongsheng'}

flask marshal return a dict

In [7]:
from flask_restplus import marshal_with, fields

In [8]:
lastname = {'last name': fields.String}

@marshal_with(lastname)
def myname():
    return {'last name':'Dongsheng', 'first name':'XI'}

In [9]:
myname()

{'last name': 'Dongsheng'}

In [10]:
type(myname()), getsizeof(myname())

(dict, 240)